app/lib/actions/katello/katello_applicability/generate_host_applicability.rb

```diff
@@ -0,0 +1,69 @@
+module Actions
+  module Katello
+    module KatelloApplicability
+      class GenerateHostApplicability < Actions::EntryAction
+        input_format do
+          param :host_id, Integer
+        end
+
+        def run
+          content_facet = ::Host.find(input[:host_id]).content_facet
+
+          content_uuids = ::Katello::Rpm.joins("inner join katello_installed_packages on \
+                               katello_rpms.name = katello_installed_packages.name and \
+                               katello_rpms.evr > katello_installed_packages.evr").
+                               joins("INNER JOIN katello_host_installed_packages ON \
+                                     katello_installed_packages.id = katello_host_installed_packages.installed_package_id \
+                                     WHERE katello_host_installed_packages.host_id = #{input[:host_id]}").pluck(:pulp_id)
```

Comment:
couldn't you just grab the rpm id directly?  instead of converting to pulp_ids and then converting back to rpm ids on line 38?

Reply:
I see what you mean.  I'll take a look at that.